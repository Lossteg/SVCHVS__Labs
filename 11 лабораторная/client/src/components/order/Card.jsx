import React from "react";
import {
  Card as MuiCard,
  CardContent,
  Typography,
  Chip,
  Paper,
} from "@mui/material";

const Card = ({ order }) => {
  const { service, properties, quantity, totalPrice } = order;

  return (
    <MuiCard
      sx={{
        maxWidth: 400,
        margin: "10px",
        padding: "10px",
        display: "flex",
        flexDirection: "column",
      }}
    >
      <CardContent>
        <Typography variant="h6">{service} Order</Typography>
        <Paper
          sx={{
            display: "flex",
            flexDirection: "row",
            padding: "10px",
            justifyContent: "space-between",
          }}
        >
          <div>
            <Typography variant="subtitle1">Properties:</Typography>
            {Object.entries(properties).map(([key, value]) => (
              <Chip
                key={key}
                label={`${key}: ${value}`}
                size="small"
                sx={{ margin: "2px" }}
              />
            ))}
          </div>
          <div>
            <Typography variant="subtitle1">Quantity:</Typography>
            <Typography variant="body1">{quantity}</Typography>
          </div>
          <div>
            <Typography variant="subtitle1">Total Price:</Typography>
            <Typography variant="body1">${totalPrice}</Typography>
          </div>
        </Paper>
      </CardContent>
    </MuiCard>
  );
};

export default Card;

