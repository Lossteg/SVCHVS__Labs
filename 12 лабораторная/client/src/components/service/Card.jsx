import React from "react";
import {
  Card as MuiCard,
  CardContent,
  CardActions,
  Button,
  Typography,
} from "@mui/material";

const Card = ({  name, imageUrl, onDetailsClick, onMakeOrderClick }) => {
  return (
    <MuiCard
      sx={{
        maxWidth: 300,
        margin: "0 5 3 0",
        alignSelf: "center",
        padding: "2px",
      }}
    >
      <img
        src={imageUrl}
        alt={name}
        style={{ width: "300px", height: "280px" }}
      />
      <CardContent>
        <Typography variant="h6">{name}</Typography>
      </CardContent>
      <CardActions style={{ display: "flex", justifyContent: "space-between" }}>
        <Button size="small" onClick={onDetailsClick}>
          Read More
        </Button>
          <Button size="small" color="primary" variant="contained" onClick={onMakeOrderClick}>
            Order
          </Button>
      </CardActions>
    </MuiCard>
  );
};

export default Card;

