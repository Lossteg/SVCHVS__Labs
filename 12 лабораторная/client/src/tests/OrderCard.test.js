import React from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import Card from "../components/order/Card";
import { MemoryRouter } from "react-router-dom";

describe("CardOrder component", () => {
  const order = {
    service: "Test Service",
    properties: { prop1: "value1", prop2: "value2" },
    quantity: 3,
    totalPrice: 150,
  };

  it("renders Card component with order details", () => {
    const { getByText } = render(<Card order={order} />);

    expect(getByText(`${order.service} Order`)).toBeInTheDocument();

    Object.entries(order.properties).forEach(([key, value]) => {
      expect(getByText(`${key}: ${value}`)).toBeInTheDocument();
    });

    expect(getByText(`Quantity:`)).toBeInTheDocument();
    expect(getByText(`${order.quantity}`)).toBeInTheDocument();

    expect(getByText(`Total Price:`)).toBeInTheDocument();
    expect(getByText(`$${order.totalPrice}`)).toBeInTheDocument();
  });
  it("OrderCard component matches snapshot", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Card order={order} />
        </MemoryRouter>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
