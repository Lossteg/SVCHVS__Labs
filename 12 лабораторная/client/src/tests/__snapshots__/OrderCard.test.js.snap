// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardOrder component OrderCard component matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1quhopn-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <h6
      className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Test Service
       Order
    </h6>
    <div
      className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-u0jpu7-MuiPaper-root"
    >
      <div>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Properties:
        </h6>
        <div
          className="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-c4zia8-MuiChip-root"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
        >
          <span
            className="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            prop1: value1
          </span>
        </div>
        <div
          className="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-c4zia8-MuiChip-root"
          onKeyDown={[Function]}
          onKeyUp={[Function]}
        >
          <span
            className="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
          >
            prop2: value2
          </span>
        </div>
      </div>
      <div>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Quantity:
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          3
        </p>
      </div>
      <div>
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          Total Price:
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          $
          150
        </p>
      </div>
    </div>
  </div>
</div>
`;
